web-demo-app/src/app/board/submodule.ts

```diff
@@ -0,0 +1,30 @@
+import { BoardModule } from './board.module';
+import { Credentials } from '../credentials';
+import { Session } from '@closerplatform/closer-sdk';
+import { SpinnerClient } from '@swagger/spinner';
+
+export abstract class SubModule {
+  public abstract readonly NAME: string;
+  protected inner: JQuery;
+
+  constructor (public boardModule: BoardModule, public credentials: Credentials) { }
```

Comment:
Skoro wartości konkstruktora nie są wymagane, to nie potrzebujemy go w klasie abstrakcyjnej

Reply:
credentials i boardModule są używane w klasach dziedziczących, a konstruktor jest tutaj, żeby nie musieć go przepisywać do wszystkich klas dziedziczących